### PR TITLE
docs: fix document error of persist with hydration

### PR DESCRIPTION
**Pull request by arvinxx**
## Related Issues or Discussions

Fixes document error

## Check List

- [ ] `yarn run prettier` for formatting code and docs

### CODE REVIEW

File: docs/integrations/persisting-store-data.md [hunk 1]
```diff
@@ -291,12 +291,12 @@ export const useBoundStore = create(
 ```
 
 ```tsx
-export function StoreConsumer(){
-  const store = useBoundStore();
+import { useBoundStore } from './path-to-store';
 
+export function StoreConsumer(){
```
Comment (sewera) [inline docs/integrations/persisting-store-data.md:296]: I don't know if Prettier does it, but a minor issue with formatting:

```suggestion
export function StoreConsumer() {
```

File: docs/integrations/persisting-store-data.md [hunk 2]
```diff
@@ -291,12 +291,12 @@ export const useBoundStore = create(
...
   // hydrate persisted store after on mount
   useEffect(() => {
-    store.persist.reHydrate();
+    useBoundStore.persist.reHydrate();
```
Comment (sewera) [inline docs/integrations/persisting-store-data.md:299]: TS: "Property 'reHydrate' does not exist on type (...). Did you mean 'rehydrate'?"

```suggestion
    useBoundStore.persist.rehydrate();
```

Comment (sewera) [reply docs/integrations/persisting-store-data.md:299]: Please also fix this typo in line 276.
"and it is left up to you to manually call `reHydrate()`."

**Review by sewera — CHANGES_REQUESTED** [review]

File: docs/integrations/persisting-store-data.md (hunk 2 above)
Comment (arvinxx) [reply docs/integrations/persisting-store-data.md]: resloved

**Review by sewera — APPROVED** [review]

**Pull request merged**